lib/koala/api/graph_api.rb

```diff
@@ -170,17 +170,23 @@ def delete_connections(id, connection_name, args = {}, options = {}, &block)
       #
       # @param options options for Facebook (see #get_object).
       #                        To get a different size photo, pass :type => size (small, normal, large, square).
+      #                        To get a object data, pass :redirect => false.
       # @param block (see Koala::Facebook::API#api)
       #
       # @note to delete photos or videos, use delete_object(id)
       #
-      # @return the URL to the image
+      # @return the URL to the image or a hash of object data if :redirect => false is passed
       def get_picture(object, args = {}, options = {}, &block)
-        # Gets a picture object, returning the URL (which Facebook sends as a header)
-        resolved_result = graph_call("#{object}/picture", args, "get", options.merge(:http_component => :headers)) do |result|
-          result ? result["Location"] : nil
+        if args[:redirect] == false
```

Comment:
What happens if args[:redirect] is nil, as it will be most of the time? Looks like it'd fall through to the other case inadvertently.